marimo/_runtime/runtime.py

```diff
@@ -1136,18 +1137,19 @@ def mutate_graph(
     async def _run_cells(self, cell_ids: set[CellId_t]) -> None:
         """Run cells and any state updates they trigger"""
 
-        # This patch is an attempt to mitigate problems caused by the fact
-        # that in run mode, kernels run in threads and share the same
-        # sys.modules. Races can still happen, but this should help in most
-        # common cases. We could also be more aggressive and run this before
-        # every cell, or even before pickle.dump/pickle.dumps()
-        with patches.patch_main_module_context(self._module):
-            while cell_ids := await self._run_cells_internal(cell_ids):
-                LOGGER.debug("Running state updates ...")
-                if self.lazy() and cell_ids:
-                    self.graph.set_stale(cell_ids, prune_imports=True)
-                    break
-            LOGGER.debug("Finished run.")
+        with run_id_context():
```

Comment:
Curious why you decided to use a context manager to get run_id rather than pass a parameter to the function. I can see that it could bloat the function definitions.

Reply:
i usually prefer args over this sideways injection. but there is a lot of layers that would need changes to pass this down. if context becomes buggy or hard to test, we can move to args later